### PR TITLE
Fix race condition related to getFieldData in lazy nodes

### PR DESCRIPTION
**Pull request by prybicki**

### CODE REVIEW

File: src/CacheManager.hpp [hunk 1]
```diff
@@ -48,6 +48,15 @@ struct CacheManager
 		cacheAge.erase(key);
 	}
 
+	std::vector<Key> getKeys() {
```
Comment (PawelLiberadzki) [inline src/CacheManager.hpp:51]: This method may be const.

File: src/api/apiCore.cpp [hunk 2]
```diff
@@ -535,8 +535,15 @@ void TapePlayer::tape_graph_get_result_size(const YAML::Node& yamlNode)
 		&out_count,
 		&out_size_of);
 
-	if (out_count != yamlNode[2].as<int32_t>()) RGL_WARN("tape_graph_get_result_size: out_count mismatch");
-	if (out_size_of != yamlNode[3].as<int32_t>()) RGL_WARN("tape_graph_get_result_size: out_size_of mismatch");
+	auto tape_count = yamlNode[2].as<int32_t>();
```
Comment (PawelLiberadzki) [inline src/api/apiCore.cpp:538]: Naming convention seem to not match a lot of code in this file: snake -> camel?
+
Also, tape_count and tape_size_of may be const.

**Comment by prybicki** [comment]
Fixed/answered, waiting for approve.

File: src/api/apiCore.cpp (hunk 2 above)
Comment (PawelLiberadzki) [reply src/api/apiCore.cpp:538]: Regarding the naming convention, understandable. In terms of const correctness, it is rather not a matter of performace, that is right. It is more about semantics - if something should not be mutable, we simply make it const.

**Review by PawelLiberadzki — APPROVED** [review]

**Pull request merged**